src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -329,19 +327,24 @@ export class MeetingManager implements AudioVideoObserver {
       console.log('[MeetingManager audioVideoDidStop] Meeting ended for all');
       this.meetingStatus = MeetingStatus.Ended;
       this.publishMeetingStatus();
+      this.leave();
+    } else if (sessionStatusCode === MeetingSessionStatusCode.Left) {
+      console.log('[MeetingManager audioVideoDidStop] Left the meeting');
+      this.meetingStatus = MeetingStatus.Left;
+      this.publishMeetingStatus();
     } else if (sessionStatusCode === MeetingSessionStatusCode.AudioJoinedFromAnotherDevice) {
       console.log('[MeetingManager audioVideoDidStop] Meeting joined from another device');
       this.meetingStatus = MeetingStatus.JoinedFromAnotherDevice;
       this.publishMeetingStatus();
+      this.leave();
     } else {
       console.log(`[MeetingManager audioVideoDidStop] session stopped with code ${sessionStatusCode}`);
+      this.leave();
```

Comment:
We should publish some status code here like `MeetingSession.Failed` if the it is a failure. We can check if failure using `sessionStatus.isFailure()` or  `sessionStatus.isAudioConnectionFailure()`. Else, add a MeetingStatus as Unknow if not related to this. This is quick fix so up to you if you want to add this now. I think at least the Failed should be published here.

Reply:
Yeah but if Michael is pressed for time then we could come back to this later :). It is a bit out of scope for this PR :)